### ISSUE TITLE
Wallet crashes when received a transaction

### ISSUE THREAD

**Issue by philipr-za** [tari-project/wallet-android#184]
When sending a transaction from a base node wallet to the android application it promptly crashes with the following crash log:

```
D/FFI: 2020-03-21 14:01:01.979376000] wallet::transaction_service::service] INFO  Transaction with TX_ID = 9585346904375281867 received from fa73fd78f28d05f10d4e45f2dd844fa1a2a131c558e84984fc8b793805a27a7c. Reply Sent
D/FFI: 2020-03-21 14:01:01.979995000] wallet::transaction_service::callback_handler] DEBUG Calling Received Transaction callback function for TxId: 9585346904375281867
I/PRETTY_LOGGER: ┌────────────────────────────────────────────────────────────────────────────────────────────────────────────────
    │ Thread: Thread-12
    ├┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄
    │ FFIWallet.onTxReceived  (FFIWallet.kt:323)
    ├┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄
I/PRETTY_LOGGER: │ Tx received. Pointer: 136375177989568
    └────────────────────────────────────────────────────────────────────────────────────────────────────────────────
D/PRETTY_LOGGER: ┌────────────────────────────────────────────────────────────────────────────────────────────────────────────────
    │ Thread: Thread-12
    ├┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄
    │ FFIWallet.onTxReceived  (FFIWallet.kt:325)
    │    WalletService.onTxReceived  (WalletService.kt:149)
    ├┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄
D/PRETTY_LOGGER: │ Tx 9585346904375281867 received.
    └────────────────────────────────────────────────────────────────────────────────────────────────────────────────
A/libc: Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x40 in tid 26577 (HeapTaskDaemon), pid 26558 (.android.wallet)
```

**Comment by philipr-za** [comment]
So I reproduced this on the Simulator so I could pull the Tombstone of the event which is attached (I gave it the .log suffix so Github will let me upload it)
[tombstone_10.log](https://github.com/tari-project/wallet-android/files/4365443/tombstone_10.log)

**Comment by kukabi** [comment]
This should be fixed by @StriderDM's #211 . Closing it, let's reopen if it persists.
